src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -779,13 +779,14 @@ private void AddTypeParameterConstraints(ImmutableArray<ITypeSymbol> typeArgumen
                             AddSpace();
 
                             bool needComma = false;
+                            var typeParameterSymbol = typeParam as TypeParameterSymbol;
 
                             //class/struct constraint must be first
                             if (typeParam.HasReferenceTypeConstraint)
                             {
                                 AddKeyword(SyntaxKind.ClassKeyword);
                                 if (format.MiscellaneousOptions.IncludesOption(SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier) &&
-                                    (typeParam as TypeParameterSymbol)?.HasNullableReferenceTypeConstraint == true) // PROTOTYPE(NullableReferenceTypes): Switch to public API when we will have one.
+                                    typeParameterSymbol?.ReferenceTypeConstraintIsNullable == true) // PROTOTYPE(NullableReferenceTypes): Switch to public API when we will have one.
                                 {
                                     AddPunctuation(SyntaxKind.QuestionToken);
```

Comment:
>QuestionToken [](start = 62, length = 13)

Not related to this PR: should we honor the IncludeNonNullableTypeModifier option here too?

Reply:
> Not related to this PR: should we honor the IncludeNonNullableTypeModifier option here too?

Yes, I am planning to add this later.

---
In reply to: [213861705](https://github.com/dotnet/roslyn/pull/29586#discussion_r213861705) [](ancestors = 213861705)